stores/metadata.go

```diff
@@ -635,23 +635,25 @@ func (s *SQLStore) ObjectEntries(ctx context.Context, path, prefix string, offse
 		return fmt.Sprintf("CONCAT(%s, %s)", a, b)
 	}
 
-	// base query
-	query := s.db.Raw(fmt.Sprintf(`SELECT SUM(size) AS size, CASE slashindex WHEN 0 THEN %s ELSE %s END AS name
+	query := s.db.Raw(fmt.Sprintf(`
+SELECT
+	SUM(size) AS size,
+	CASE slashindex
+	WHEN 0 THEN %s
+	WHEN 1 THEN %s
+	ELSE %s
+	END AS name
+FROM (
+	SELECT size, trimmed, INSTR(trimmed, "/") AS slashindex
 	FROM (
-		SELECT size, trimmed, INSTR(trimmed, ?) AS slashindex
-		FROM (
-			SELECT size, SUBSTR(object_id, ?) AS trimmed
-			FROM objects
-			WHERE object_id LIKE ?
-		) AS i
-	) AS m
-	GROUP BY name
-	LIMIT ? OFFSET ?`, concat("?", "trimmed"), concat("?", "substr(trimmed, 1, slashindex)")), path, path, "/", utf8.RuneCountInString(path)+1, path+"%", limit, offset)
-
-	// apply prefix
-	if prefix != "" {
-		query = s.db.Raw(fmt.Sprintf("SELECT * FROM (?) AS i WHERE name LIKE %s", concat("?", "?")), query, path, prefix+"%")
-	}
+		SELECT size, SUBSTR(object_id, ?) AS trimmed
+		FROM objects
+		WHERE object_id LIKE ?
+	) AS i
+) AS m
+WHERE name LIKE ?
```

Comment:
`NOTE` I query `name LIKE` even when a prefix is not given. I thought about splitting it up but it's not great for readability and the way the query is built I doubt it would hurt performance. I measured it on arequipa and it made no difference.